### PR TITLE
Adding swagger sorting command to tcp-client

### PR DESCRIPTION
**Pull request by allenjzhang**

### CODE REVIEW

**Review by timotheeguerin — APPROVED** [review]

**Comment by timotheeguerin** [comment]
Be good to see what @catalinaperalta thinks of that first too

File: tools/tsp-client/README.md [hunk 1]
```diff
@@ -44,6 +44,8 @@ Generate a client library from a TypeSpec project. The `generate` command should
 
 ### convert
 Convert an existing swagger specification to a TypeSpec project. This command should only be run once to get started working on a TypeSpec project. TypeSpec projects will need to be optimized manually and fully reviewed after conversion. When using this command a path or url to a swagger README file is required through the `--swagger-readme` flag.
+### sortSwagger
```
Comment (catalinaperalta) [inline tools/tsp-client/README.md:47]: Nit: needs a space
```suggestion

### sortSwagger
```

File: tools/tsp-client/package.json [hunk 2]
```diff
@@ -38,7 +38,8 @@
     "@types/mocha": "^10.0.1",
     "@types/node": "^20.4.8",
     "@types/prompt-sync": "^4.2.1",
-    "@typespec/compiler": "0.50.0",
+    "@typespec/compiler": "^0.58.0",
+    "@azure-tools/typespec-autorest": "^0.44.0",
```
Comment (catalinaperalta) [inline tools/tsp-client/package.json:42]: Do folks need to use v0.44.0 or later to be able to run this command?

File: tools/tsp-client/src/index.ts [hunk 3]
```diff
@@ -355,6 +362,16 @@ async function main() {
           }
         }
         break;
+    case "sortSwagger":
+        Logger.info("Sorting a swagger content...");
+        let swaggerFile =  options.swaggerFile;
+        if (swaggerFile === undefined || await !doesFileExist(swaggerFile)) {
+            throw new Error(`Swagger file not found: ${swaggerFile??"[Not Specified]"}`);
+        }
+        swaggerFile = normalizePath(resolve(swaggerFile));
```
Comment (catalinaperalta) [inline tools/tsp-client/src/index.ts:371]: Should we be doing this before we do the doesFileExist check on line 368?

File: tools/tsp-client/src/log.ts [hunk 4]
```diff
@@ -68,6 +68,7 @@ Commands:
   generate    Generate from a TypeSpec project                          [string]
   update      Sync and generate from a TypeSpec project                 [string]
   convert     Convert a swagger specification to TypeSpec               [string]
+  sortSwagger Sort a swagger specification based on TypeSpec ordering   [string]
```
Comment (catalinaperalta) [inline tools/tsp-client/src/log.ts:71]: Can we rename this to `sort-swagger` ?

File: tools/tsp-client/src/options.ts [hunk 5]
```diff
@@ -136,6 +138,15 @@ export async function getOptions(): Promise<Options> {
         process.exit(1);
       }
     }
+
+    if (command === "sortSwagger") {
```
Comment (catalinaperalta) [inline tools/tsp-client/src/options.ts:142]: Nit: can we rename to `sort-swagger`?

**Review by catalinaperalta — COMMENTED** [review]
Can we also add a test for this scenario?

File: tools/tsp-client/package.json (hunk 2 above)
Comment (allenjzhang) [reply tools/tsp-client/package.json:42]: sorting API is introduced recently into autorest.

File: tools/tsp-client/src/index.ts (hunk 3 above)
Comment (allenjzhang) [reply tools/tsp-client/src/index.ts:371]: Seeing the above code, it was done this way. I am not sure if this does anything as if `doesFileExist` works with the path, `readFile` should work as well. But I can adjust.

File: tools/tsp-client/package.json (hunk 2 above)
Comment (catalinaperalta) [reply tools/tsp-client/package.json:42]: Makes sense, then we should also have it under dependencies.

Comment (timotheeguerin) [reply tools/tsp-client/package.json:42]: Ah yeah should be in dependencies not devDependencies

Comment (catalinaperalta) [reply tools/tsp-client/package.json:42]: Also, I think we should float peerDependencies for typespec/compiler between 0.58.0 - 1.0.0

Comment (allenjzhang) [reply tools/tsp-client/package.json:42]: ^0.44.0 is `0.44.0 <= ver < 1.0.0`

Comment (timotheeguerin) [reply tools/tsp-client/package.json:42]: no for `0.` `^` doesn't allow next minors `^0.44.0` its just `>=0.44.0 && < 0.45.0`.

However in this case its not meant to be used as a peerDependency but as a regular dependency so this is correct.
This dependency can be updated whenever(probably will complain about the version of the compiler if yhou don't update) but in theory it doesn't matter its just loading the function from this lib

Comment (catalinaperalta) [reply tools/tsp-client/package.json:42]: Also, the comment was pointing out that these deps should be under "dependency"

Comment (catalinaperalta) [reply tools/tsp-client/package.json:42]: We float typespec/compiler in peerDependencies, which is where I was saying we need to update it to 0.58.0

Comment (catalinaperalta) [reply tools/tsp-client/package.json]: I'll fix the peer deps in a separate pr @allenjzhang. Resolving now

Comment (catalinaperalta) [reply tools/tsp-client/package.json]: https://github.com/Azure/azure-sdk-tools/issues/8766 tracking issue before next tsp-client release

**Review by catalinaperalta — APPROVED** [review]

**Comment by catalinaperalta** [comment]
FYI are you aware of this issue: https://github.com/Azure/autorest/issues/4944 ? Should it be closed now or is it a separate effort?

**Review by catalinaperalta — APPROVED** [review]

**Pull request merged**